### ISSUE TITLE
Question/bug? about assume negative loss function

### ISSUE THREAD

**Issue by kopop8** [elijahcole/single-positive-multi-label#12]
If i have read the paper and code correctly. Unobserved labels should be labelled as -1 and observed labels should be 1.  In the function 'loss_an' it checks for labels that are zero while in the paper the formula says !=1 which is a bug in the code I think as it will only consider positive observed samples causing each label to always be activated while training.

![image](https://user-images.githubusercontent.com/28300635/228241624-92be4e7a-c49c-4fe9-a98c-5a81c190e83d.png)

![image](https://user-images.githubusercontent.com/28300635/228241689-5c11a2d1-0d1b-4263-9b23-d87eed10d6ee.png)

**Comment by elijahcole** [comment]
Thanks for the question! 

The notation in the paper does differ from the implementation in the code. I think the confusion stems from a (probably sub-optimal) choice I made to use different label representations for fully observed losses and SPML losses in the code. 

For fully observed losses like $\mathcal{L}_\mathrm{BCE}$ we use +1 for positive and 0 for negative. To see how we format fully observed label matrices, see e.g. [here](https://github.com/elijahcole/single-positive-multi-label/blob/main/preproc/format_pascal.py#L55). 

For partially observed losses like $\mathcal{L}_\mathrm{AN}$ we use +1 for positive, -1 for negative, and 0 for unobserved. For SPML in particular we expect one label to be positive (+1) and the rest to be unobserved (0). We should not have any observed negative labels (-1). To see how we convert fully observed label matrices to observed label matrices, see [here](https://github.com/elijahcole/single-positive-multi-label/blob/main/preproc/generate_observed_labels.py#L73). 

Please re-open this issue if I have not answered your question!

**Comment by kopop8** [comment]
Thank you so much for the quick response!  This actually helps me a lot thank you
